Add fluentd to logging-agent DaemonSet

christianberg opened the pull request:
Add a fluentd container to the logging-agent DaemonSet, with fluentd
config stored in a ConfigMap.

christianberg committed:
Add fluentd to logging-agent DaemonSet (WIP)

Add a fluentd container to the logging-agent DaemonSet, with fluentd
config stored in a ConfigMap.

Signed-off-by: Christian Berg <christian.berg@zalando.de>

christianberg commented:
Work in progress, do not merge.

mikkeloscar commented:
~@christianberg can you please push to a branch in this repository instead of a fork? This will allow us to run e2e tests once this is ready (we can't run e2e on a fork right now). I gave you write permission to the repo. :)~

Actually, since this is using the logging-agent daemonset which we don't e2e test right now anyway, it probably doesn't matter so much.. :)

christianberg commented:
Of course! I'll open a new PR and close this one.

christianberg committed:
Fix fluentd image version

Signed-off-by: Christian Berg <christian.berg@zalando.de>

christianberg committed:
Make bucket name configurable

Signed-off-by: Christian Berg <christian.berg@zalando.de>

christianberg committed:
Allow pod to access S3

Signed-off-by: Christian Berg <christian.berg@zalando.de>

christianberg committed:
Fix pod variable in S3 object path

Signed-off-by: Christian Berg <christian.berg@zalando.de>

christianberg committed:
Add GetObject permission to LoggingAgent IAM role

Signed-off-by: Christian Berg <christian.berg@zalando.de>

christianberg committed:
Fluentd config: simplify buffer variables

Signed-off-by: Christian Berg <christian.berg@zalando.de>

christianberg committed:
Tune S3 Log Shipping Parameters

Signed-off-by: Christian Berg <christian.berg@zalando.de>

christianberg committed:
Store fluentd pos_file on hostPath volume

Signed-off-by: Christian Berg <christian.berg@zalando.de>

christianberg committed:
Pass cluster environment to kube-log-watcher

Signed-off-by: Christian Berg <christian.berg@zalando.de>

christianberg committed:
Add 'age' filter to Fluentd

This drops any log events older than 4 hours. Age limit is currently
hard-coded in the custom plugin, will be configurable later.

Signed-off-by: Christian Berg <christian.berg@zalando.de>

christianberg committed:
Add Fluentd metrics endpoint

Using Fluentd prometheus plugin.

Signed-off-by: Christian Berg <christian.berg@zalando.de>

christianberg committed:
Add feature toggle for fluentd

Signed-off-by: Christian Berg <christian.berg@zalando.de>

christianberg committed:
Bump versions of kube-log-watcher and fluentd

Signed-off-by: Christian Berg <christian.berg@zalando.de>

christianberg committed:
Set default S3 logging bucket name

Signed-off-by: Christian Berg <christian.berg@zalando.de>

aermakov-zalando commented:
LGTM. Would you mind bumping the reserved CPU/memory accordingly [here](https://github.com/zalando-incubator/kubernetes-on-aws/blob/dev/cluster/config-defaults.yaml#L6)? We sadly don't have this automated yet. No need to do it conditionally, it's fine to just bump it for all clusters.

christianberg committed:
Reduce resource usage for logging-agent

Make the CPU and Memory requests/limits for the three logging-agent
containers configurable, and reduce the default values.

Signed-off-by: Christian Berg <christian.berg@zalando.de>

aermakov-zalando commented on cluster/manifests/logging-agent/daemonset.yaml:
```diff
@@ -105,14 +108,18 @@ spec:
           limits:
-            cpu: 50m
-            memory: 250Mi
+            cpu: "{{ index .ConfigItems "logging_watcher_cpu" }}"
```
Please just use `.ConfigItems.logging_watcher_cpu`.

aermakov-zalando commented:
Closed in favor of #1826.
